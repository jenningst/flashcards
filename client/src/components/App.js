import React, { useState } from 'react';
import styled from 'styled-components'
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { darkTheme, lightTheme } from '../themes/theme';
import { PackProvider } from '../contexts/pack-context';

import { ApolloClient, HttpLink } from 'apollo-boost';
import { ApolloProvider } from 'react-apollo';
import { InMemoryCache } from 'apollo-cache-inmemory';

import Dashboard from './Dashboard';
import ComposePack from './ComposePack';
import PackContainer from '../containers/PackContainer';

const apolloClient = new ApolloClient({
  link: new HttpLink({ uri: "http://localhost:4000/graphql" }),
  cache: new InMemoryCache(),
});

function App() {
  const [isDarkMode, setIsDarkMode] = useState(false);

  return (
    <ApolloProvider client={apolloClient}>
      <PackProvider>
        <ThemeProvider theme={isDarkMode ? darkTheme : lightTheme}>
          <AppWrapper className="App">
            <Router>
              <Route path="/" exact component={Dashboard} />
              <Route path="/create-pack" component={ComposePack} />
              <Route path="/pack/:id" component={PackContainer} />
            </Router>
          </AppWrapper>
        </ThemeProvider>
      </PackProvider>
    </ApolloProvider>
  );
};

const AppWrapper = styled.div`
  height: 100vh;
  width: 100vw;
  background: ${props => props.theme.color.main.offWhite};
`;

export default App;
