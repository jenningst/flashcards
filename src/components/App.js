import React, { useContext } from 'react';
import { PackProvider } from '../contexts/packContext';
import ThemeContext, { ThemeProvider } from '../contexts/themeContext';


import AppRouter from '../containers/AppRouter';

function App() {
  const theme = useContext(ThemeContext);

  return (
    <PackProvider>
      <ThemeProvider value={theme}>
        <AppRouter />
      </ThemeProvider>
    </PackProvider>
  );
};

export default App;
