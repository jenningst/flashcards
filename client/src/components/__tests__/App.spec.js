import React from 'react';
import { shallow } from 'enzyme';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import App from '../App';

afterEach(cleanup);

describe('<App /> spec', () => {
  it('assert it renders correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('assert primary div is present', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('div.app')).toHaveLength(1);
  });
});
