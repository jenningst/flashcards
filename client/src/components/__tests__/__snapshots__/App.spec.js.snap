// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> spec assert component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
    style="height: 100vh; width: 100vw;"
  >
    <div
      class="Dashboard sc-gZMcBi ffKCQX"
    >
      <header
        class="Dashboard__header sc-gqjmRU hGzDjA"
      >
        <svg
          class="Dashboard__button-menu sc-VigVT fCGeOx"
          data-testid="button-menu"
        >
          menu.svg
        </svg>
      </header>
      <section
        class="Dashboard__list sc-jTzLTM FholT"
      >
        <button
          class="Dashboard__button-create-pack sc-fjdhpX ddgael"
          data-testid="create-pack-button"
          type="button"
        >
          <h1
            class="sc-bdVaJa beESmM"
          >
            +
          </h1>
        </button>
        Loading...
      </section>
    </div>
  </div>
</DocumentFragment>
`;
